agent-install/agent-install.sh

```diff
@@ -2413,7 +2413,7 @@ function update_cluster() {
     set -e
 
     if is_agent_registered ; then
-        if [[ "$IS_HORIZON_DEFAULTS_CORRECT" != "true" ]] || ! is_registration_correct ; then
+        if [[ "$IS_HORIZON_DEFAULTS_CORRECT" != "true" ]] || ! is_registration_correct || [[ "$IS_HORIZON_ORG_ID_SAME" == "false" ]]; then
```

Comment:
Shouldn't we instead have `is_registration_correct()` check if the org is different for both devices and edge clusters?